elastic/fleet-server
# Allow to compile a binary for debug

## What is the problem this PR solves?

The generated binary cannot be debugged as it omits debug symbol tables 

## How does this PR solve the problem?

It adds a `DEV` environment variable read by Make which when set to true (actually, not empty) it preserves the debug symbol tables, remove compiler optimizations and inlining.

## How to test this PR locally

 - Compile with DEV=true:
```
DEV=true SNAPSHOT=true make release-linux/amd64 # adjust for your platform
```
 - extract the tar ball
 - check the generated binary 
```shell
objdump --dwarf path/to/your/binary | grep -A 7 'DW_TAG_compile_unit' | grep 'DW_AT_name\|DW_AT_producer' | grep -C 2 elastic
```

the output should contain lines similar to `Go cmd/compile go1.18.2; -N -l -shared regabi`, like these:

```
    <56f9be>   DW_AT_name        : github.com/elastic/fleet-server/v7/internal/pkg/checkin
    <56fa09>   DW_AT_producer    : Go cmd/compile go1.18.2; -N -l -shared regabi
    <57015a>   DW_AT_name        : github.com/elastic/fleet-server/v7/internal/pkg/throttle
    <5701a6>   DW_AT_producer    : Go cmd/compile go1.18.2; -N -l -shared regabi
```
## Checklist





## Related issues

N/A
